packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -1237,6 +1250,16 @@ export class UserPool extends UserPoolBase {
       }
       if (props.mfaSecondFactor!.otp) {
         enabledMfas.push('SOFTWARE_TOKEN_MFA');
+      } if (props.mfaSecondFactor!.email) {
```

Comment:
An idea to make this code more readable - we could add a helper function to validate email MFA, something like:

```ts
function validateEmailMfa(props: UserPoolProps) {
  if (!props.email || this.emailConfiguration?.emailSendingAccount !== 'DEVELOPER') {
    throw new Error('To enable email-based MFA, set `email` property to the Amazon SES email-sending configuration.');
  }
  
  if (props.advancedSecurityMode === AdvancedSecurityMode.OFF) {
    throw new Error('To enable email-based MFA, set `advancedSecurityMode` to `AdvancedSecurity.ENFORCED` or `AdvancedSecurity.AUDIT`.');
  }
}
```

and then replace the if statements with:

```ts
if (props.mfaSecondFactor!.otp) {
  enabledMfas.push('SOFTWARE_TOKEN_MFA');
}

if (props.mfaSecondFactor!.email) {
  validateEmailMfa(props);
  enabledMfas.push('EMAIL_OTP');
}
```

Makes it easier to follow this code

Reply:
Thanks. I've updated. 
I thought private methods would be better so I went with that approach.

Reply:
Looks good, thanks @mazyu36 !